### DISCUSSION TITLE
Installation problem rpi 3+ / bookworm / HQcam

### DISCUSSION THREAD

**Discussion by gorante** [aaronwmorris/indi-allsky#1371] [Q&A] [answered]
Hi all, I'm trying to install indi allsky on a Raspberry 3+, using a Raspberry HQ cam (IMX477). 
First of alll: this setup is working without issues with another Allsky software, I'm just swapping the SD card.

The problem: at the end of the setup script i get this message:

"The DBUS user session is not defined
Now that the dbus package has been installed...
Please reboot your system and re-run this script to continue"

I reboot, relaunch the script and i get the same error.

If i try to launch indi-allsjy with "systemctl --user start indi-allsky" i get this error: "Failed to connect to bus: No medium found"

What I did:

- installed and fully updated latest 64bit bookworm version of Raspberry Pi OS
- launched setup script (choosing: 2-Libcamera and 1-Libcamera_imx477)
- compiled indilib drivers
- relaunched setup script

I tried also with 64bit bullseye version: same problem
I also tried to compile indilib from another source: same problem

Am I missing something? Anyone have some idea?
Thanks!

**Comment by aaronwmorris** [comment]
With the user that is supposed to run indi-allsky, can you show the output of the following commands:

```
systemctl --user status dbus.service

systemctl --user status dbus.socket
```

**Reply by gorante** [reply]
Hi Aaron, the output is "Failed to connect to bus: No medium found"

<img width="972" alt="Screenshot 2024-07-02 alle 17 45 33" src="https://github.com/aaronwmorris/indi-allsky/assets/4375800/e3f51325-2776-4182-a7ee-dbff2212dd4f">

the camera is connected and working (I get images also with rpicam-jpeg)

<img width="965" alt="Screenshot 2024-07-02 alle 17 47 34" src="https://github.com/aaronwmorris/indi-allsky/assets/4375800/8802f541-5cd5-4e16-b89c-96493129c535">

Thanks!

**Reply by aaronwmorris** [reply]
I am stumped.  Are you just logged in via SSH?  In the console, or via the GUI?  Anything that might help me understand?

**Reply by gorante** [reply]
> I am stumped. Are you just logged in via SSH? In the console, or via the GUI? Anything that might help me understand?

I'd say that I'm pretty stumped too 😄 
By the way: yes I'm logged in through SSH, via console (what you see in the screenshot it's iTerm on my macbook).

I've done some tests, always with the same results:
- first I started from a raspberry pi os bookworm lite image
- then I tried with a bullseye lite image
- then I tried with bookworm full (with GUI) image, hoping that the problem was caused by missing libraries or stuff

All I've got was always the same kind of error, nothing changed.

Installation process:

- flash sdcard with raspberry pi imager
- login via ssh
- use raspi-config to enable all connections (not really needed, but avoid any connection issue)
- apt update && apt upgrade 
- install byobu (this is to keep ssh sessions up also if disconnected
- move to home user folder and start the installation process as in the guide here: https://github.com/aaronwmorris/indi-allsky/wiki/Getting-Started

That's it!

**Comment by gorante** [comment]
@aaronwmorris  I guess no idea about the problem came to mind.

In any case: I got a RaspberryPi5 and tried to do the installation procedure there, result: same error.
In this case, however, I don't have the HQ cam connected but a ZWO Asi 294 (which I don't know if it is supported by indi-allsky, it certainly is by INDIlib).
To connect the HQ cam I have to find a suitable flat cable because the RPI5 uses smaller ones.

To try to identify the problem and reduce the cases of errors I would like to understand one thing: if the camera is not connected, indi-allsky starts anyway but does not show images or does it give errors? Could that "Failed to connect to bus: No medium found" error be related to a camera reading error?

**Reply by aaronwmorris** [reply]
I am working on this today, I am finally in my office where I can test things.  I am wondering if it might be due to the Raspberry Pi Imager program.

**Reply by gorante** [reply]
> I am working on this today, I am finally in my office where I can test things. I am wondering if it might be due to the Raspberry Pi Imager program.

Thanks, well good point i could try flashing with another imager.

**Reply by aaronwmorris** [reply]
I have only recently started using the Imager program myself, and I do not think I have tested it with an indi-allsky build yet.  I normally just `dd` images to the sdcard and configure from the console (like a savage :-) ).

I do not think the Imager program is likely the cause of any problems.  If there is anything unique to your setup or your normal workflow, please let me know.

Which image do you normally pick in the Imager program?

**Reply by gorante** [reply]
> I have only recently started using the Imager program myself, and I do not think I have tested it with an indi-allsky build yet. I normally just `dd` images to the sdcard and configure from the console (like a savage :-) ).
> 
> I do not think the Imager program is likely the cause of any problems. If there is anything unique to your setup or your normal workflow, please let me know.
> 
> Which image do you normally pick in the Imager program?

I'm testing anyways starting with an image flashed with Balena Etcher (next i'll do a dd, surely :D )
by the way i'm using raspberry os 64 bit bookworm lite version, but i tested it also using a full version (with GUI).

Anything unique... I don't think so, I've listed the steps I take before and there's nothing more.

**Reply by gorante** [reply, marked as answer]
@aaronwmorris I may have found one issue.
Looks like something doesn't like to run on screen (byobu) session.
I've just launched .setup.sh script out of byobu session and it started installing stuff 

<img width="925" alt="Screenshot 2024-07-05 alle 21 14 17" src="https://github.com/aaronwmorris/indi-allsky/assets/4375800/2ed93525-8c47-418f-b6a9-d0df8d66fffe">

until it crashed badly here.

<img width="1250" alt="Screenshot 2024-07-05 alle 21 16 50" src="https://github.com/aaronwmorris/indi-allsky/assets/4375800/2c7e57cd-82ca-41a4-8ae6-d0c3b73562fb">

That's pretty odd though, never had a problem running stuff via SSH and on screen session in... 20 years or more.

Now I'm slowly proceeding, relaunching setup script and crashing, relaunching crashing but going further.

**Reply by aaronwmorris** [reply]
I have tested both with bookwork 12 64-bit full and the lite version.  No issue with either.

After re-reading your post, I started looking at byobu a little while ago.  I have never noticed a problem in screen, but I have never used byobu.  It kind of makes sense that this can cause a problem.

If something is causing byobu not to inherit the environment variables from the parent session, then systemctl commands will fail because it does not know how to communicate with the DBUS user session socket.  Or perhaps, the user session socket is not being created for some reason.

The download errors for the python modules is an occasional problem.  Its probably a problem with the mirror you are hitting, or an issue with your internet connection.

You can try purging the wheel cache and restarting...

```
source virtualenv/indi-allsky/bin/activate

pip cache purge
```

**Reply by gorante** [reply]
> After re-reading your post, I started looking at byobu a little while ago. I have never noticed a problem in screen, but I have never used byobu. It kind of makes sense that this can cause a problem.
> 
> If something is causing byobu not to inherit the environment variables from the parent session, then systemctl commands will fail because it does not know how to communicate with the DBUS user session socket. Or perhaps, the user session socket is not being created for some reason.

Yes, I've noticed that the DBUS error happens when systemctl can't create a DBUS session, so I my mind went to byobu.
And I can confirm that after some errors with piwheels it ended the setup and now it's running fine (with the 294 attached).

It's pretty strange though, i've configured systems throught byobu in years without problems (I always use it to avoid interruption of scripts or installations due to connection fails or laptops going in sleep mode...). 
There shoud be surely something that doesn't allow byobu tu get variables. I'll take a look later on this issue.

For now, problem is solved, and thanks for help!

**Reply by aaronwmorris** [reply]
I have added a note to the Getting Started page about possible problems with screen, tmux, and byobu.
